src/include/sof/audio/ipc-config.h

```diff
@@ -9,6 +9,42 @@
 #include <stdint.h>
 #include <stdbool.h>
 
+/* ACE 2.0 and newer */
+#define GTW_DMA_CONFIG_ID		0x1000
+#define GTW_DMA_INVALID_CHANNEL_ID	0xffffffff
+/* Maximum number of devices */
+#define GTW_DMA_DEVICE_MAX_COUNT 16
+
+struct dma_device_stream_channel_map {
+	uint32_t device_address;
+	uint32_t channel_map;
+};
+
+struct dma_stream_channel_map {
+	uint32_t device_count;
+	struct dma_device_stream_channel_map map[GTW_DMA_DEVICE_MAX_COUNT];
+};
+
+struct ipc_dma_config {
+	uint8_t dma_method;		/**< DMA method */
+	uint8_t pre_allocated_by_host;	/**< 1 - DMA pre-allocated by host, otherwise to be
+					  *       allocated by DSP
+					  */
+	uint16_t rsvd;
+	uint32_t dma_channel_id;	/**< channel ID if DMA pre-allocated by host, otherwise
+					  *   invalid channel ID
+					  */
+	uint32_t stream_id;
+	/* DmaStreamChannelMap */
+	struct dma_stream_channel_map channel_map;
+	uint32_t dma_priv_config_size;
+	uint8_t  dma_priv_config[0];
+} __attribute__((packed, aligned(4)));
+
+#define GET_IPC_DMA_CONFIG(buffer, size) \
+	((uint32_t *)((buffer) + ((size) * 4) - sizeof(struct ipc_dma_config)))
```

Comment:
this definition apparently *requires* that `buffer` is a pointer to 8-bit data, e.g. a `uint8_t *` pointer. Maybe it's better to make it explicit: either typecast to such a type, or make this a `static inline`. Also worth adding, that `size` is in 32-bit dwords.